compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml

```diff
@@ -146,8 +146,26 @@ instance Show (ContractId a) where
     None -> "<contract-id>"
     Some t -> t
 
--- | HIDE This is currently used in the internals of Daml script and Daml triggers
--- but not really something that we want to expose to users.
+-- | Change the underlying type of a contract id. For example, given two
+-- types `A` and `B`, and `cid : ContractId A`,
+-- `coerceContractId @A @B cid : ContractId B`.
+--
+-- This is not very useful when `A` and `B` are both templates, but it can be
+-- useful when either or both are interfaces but
+-- `fromInterfaceContractId` and `toInterfaceContractId` don't work.
```

Comment:
Would there be any value in outlining scenarios where `*InterfaceContractId` functions do not work?

Reply:
hm the only one I can think of is converting from interface `I` to interface `J` where one implies the other i.e. we know that for any template `T`, if `interface instance I for T` then `interface instance J for T`. However I don't want to rule out other scenarios that might come up - but then there's the question, could there be other scenarios?

Reply:
I'm wondering now if instead what we want is a synonym of `coerceContractId` for exactly the scenario above (while keeping `coerceContractId` hidden), something like

```
-- | used to coerce between contract ids of different interfaces, not checked, subsequent fetch/exercise might fail, etc etc
coerceInterfaceContractId : forall i j. (HasInterfaceTypeRep i, HasInterfaceTypeRep j) => ContractId j -> ContractId i
coerceInterfaceContractId = coerceContractId
```

Reply:
> However I don't want to rule out other scenarios that might come up - but then there's the question, could there be other scenarios?

Maybe we could just say something like "e.g. for any template T, if interface instance I for T then interface instance J for T"? In doing that, we point out a specific known scenario without committing to being fully inclusive?

> I'm wondering now if instead what we want is a synonym of `coerceContractId`

That's an interesting idea.

As the user might already be directed to/thinking about the `*InterfaceContractId` set of functions, I can see why you might bundle in the extra `coerceInterfaceContractId` function.

Maybe this distinction might cause extra confusion for some users?